packages/seducer/src/useSeducer.js

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import reducer from "./reducer";
+import getTypes from "./getTypes";
+
+export const useSeducer = (
+  actions,
+  initialState = {},
+  initializer = undefined,
+  hasLogger = false,
+  interceptors = {}
+) => {
+  if (!actions) {
+    throw new Error("Actions are required for useSeducer, is an object with their actions as their properties");
+  }
+
+  const types = React.useMemo(() => getTypes({ ...interceptors, ...actions }), [actions, interceptors]);
+
+  const reducerMemo = React.useMemo(() => {
+    return reducer(actions || {}, hasLogger, interceptors);
+  }, [actions, hasLogger, interceptors]);
+
+  const [state, dispatch] = React.useReducer(
+    reducerMemo,
+    initialState,
+    // if initializer is not a function just skipped by passing undefined so hasLogger can work
+    typeof initializer === "function" ? initializer : undefined
+  );
+
+  /**
+   * This is a helper that allowed to change the regular dispatch signature which is
+   * dispatch({ type: "action", payload: value })
+   * to
+   * dispatch(types.type, value) which IMO is more versatile, shorter and pleasant to look at it.
+   */
+  const memoCustomDispatch = React.useMemo(
```

Comment:
why not `useCallback` here? 

```js
const memoCustomDispatch = React.useCallback((...args) {
    const [type, payload] = args;
    dispatch({ type, payload })
}, [])
```

Reply:
Interesting, the React docs say that 
> useCallback:
> Returns a memoized callback.
> 
> And that useMemo:
> Returns a memoized value.

🤔  I guess in this case is a function and callback make sense. but is also a return value from the useMemo function, and useCallback is based on useMemo, so 🤷‍♂️, I will change it to make it more readable.

Reply:
though I would argue that 
```js
  const memoCustomDispatch = React.useMemo(
    () =>
      function dispatchCallback(...args) {
        const [type, payload] = args;
        dispatch({ type, payload });
      },
    []
  );
```

is really clear about what is happening. what do you think @allison-c @jamiek-acl

Reply:
`useMemo` is to memoize an "expensive to compute" value... you're not trying to memoize a value here.

Reply:
or for `referential equality between renders` which is what I'm doing here. (useCallback and useMemo) work for referential equality.

Reply:
I end replacing useMemo with useCallback for brevity and to keep moving this PR :D